woocommerce/woocommerce-ios
# Tracks: Add error code to networking error tracking

Closes: #8527


## Description
Most networking errors are wrapped as `AFError` since we use Alamofire to manage the Networking stack. When tracking events related to networking, since we're only checking for `NSError` types rather than `AFError` types when unwrapping and assigning track properties, some of these tracks will not contain details like the error code.

## Changes
- Adds a check to the `errorProperties` passed to `WooAnalytics.track(:)`, where we override the value for `errorCode` if the type matches `AFError`, otherwise we keep using `NSError` properties.

## Testing instructions

- Run the app and either log out from your store, create a new store, or attempt to [log into this test store](http://final-wildcat.jurassic.ninja) 
- Tap "Connect with Jetpack"
- Enter the wrong credentials
- See a `AnyHashable("error_code"): "401"` error in the console:

Example of full track event:

**Before**
```
2023-02-01 14:47:13.529831+0800 WooCommerce[82253:7598094] 
🔵 Tracked login_jetpack_site_credential_did_show_error_alert, properties: [
  AnyHashable("error_domain"): "Alamofire.AFError", 
  AnyHashable("blog_id"): 121710041, 
  AnyHashable("error_code"): "3", 
  AnyHashable("is_wpcom_store"): false, 
  AnyHashable("error_description"): "Alamofire.AFError.responseValidationFailed(reason: Alamofire.AFError.ResponseValidationFailureReason.unacceptableStatusCode(code: 401))"
]
```

**After:**
```
2023-02-01 14:36:40.578828+0800 WooCommerce[81728:7589271] 
🔵 Tracked login_jetpack_site_credential_did_show_error_alert, properties: [
  AnyHashable("error_domain"): "Alamofire.AFError", 
  AnyHashable("blog_id"): 121710041, 
  AnyHashable("is_wpcom_store"): false, 
  AnyHashable("error_code"): "401", 
  AnyHashable("error_description"): "Alamofire.AFError.responseValidationFailed(reason: Alamofire.AFError.ResponseValidationFailureReason.unacceptableStatusCode(code: 401))"
]
```

## Screenshots



---
